test/functional/pdo_sqlsrv/pdo_ae_output_param_binary_size.phpt

```diff
@@ -0,0 +1,211 @@
+        $ch0 = 'd';
+        $ch1 = 'e';
+
+        foreach ($dataTypes as $dataType) {
+            $maxtype = strpos($dataType, "(max)");
+            foreach ($lengths as $length) {
+                if ($maxtype !== false) {
+                    $type = $dataType;
+                } else {
+                    $type = "$dataType($length)";
+                }
+                trace("\nTesting $type:\n");
+                    
+                //create and populate table
+                $colMetaArr = array(new ColumnMeta($type, "c_det"), new ColumnMeta($type, "c_rand", null, "randomized"));
+                createTable($conn, $tbname, $colMetaArr);
+                $input0 = str_repeat($ch0, $length);
+                $input1 = str_repeat($ch1, $length);
+                $ord0 = convert2Hex($ch0, $length);
+                $ord1 = convert2Hex($ch1, $length);
+                insertRow($conn, $tbname, array("c_det" => new BindParamOp(1, $input0, "PDO::PARAM_LOB", 0, "PDO::SQLSRV_ENCODING_BINARY"),
+                                                "c_rand" => new BindParamOp(2, $input1, "PDO::PARAM_LOB", 0, "PDO::SQLSRV_ENCODING_BINARY")), "prepareBindParam");
+                    
+                // fetch with PDO::bindParam using a stored procedure
+                dropProc($conn, $spname);
+                $spSql = "CREATE PROCEDURE $spname (
+                                @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                $conn->query($spSql);
+                
+                // call stored  procedure
+                $outSql = getCallProcSqlPlaceholders($spname, 2);
+                foreach ($pdoParamTypes as $pdoParamType) {
+                    $stmt = $conn->prepare($outSql);
+                    trace("\nParam $pdoParamType with INOUT = $inout\n");
+                    
+                    if ($inout && $pdoParamType != PDO::PARAM_STR) {
+                        // Currently do not support getting binary as strings + INOUT param
+                        $paramType = $pdoParamType | PDO::PARAM_INPUT_OUTPUT;
+                    } else {
+                        $paramType = $pdoParamType;
+                    }
+
+                    $det = "";
+                    $rand = "";
+                    if ($pdoParamType == PDO::PARAM_STR) {
+                        $stmt->bindParam(1, $det, $paramType, $length, PDO::SQLSRV_ENCODING_BINARY); 
+                        $stmt->bindParam(2, $rand, $paramType, $length, PDO::SQLSRV_ENCODING_BINARY); 
+                    } elseif ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                        $det = $rand = 0;
+                        $stmt->bindParam(1, $det, $paramType, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+                        $stmt->bindParam(2, $rand, $paramType, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+                    } else {

```

Comment:
I thought PDO::SQLSRV_ENCODING_BINARY can be specified with PDO::PARAM_LOB

Reply:
Not sure what you mean?